pkg/policy/consumer.go

```diff
@@ -216,40 +200,43 @@ func (c *Consumable) removeFromMaps(id NumericIdentity) {
 		})
 
 		scopedLog.Debug("Updating policy BPF map: denying Identity")
-		if err := m.DeleteConsumer(id.Uint32()); err != nil {
+		if err := m.DeleteIdentity(id.Uint32()); err != nil {
 			scopedLog.WithError(err).Warn("Update of policy map failed")
 		}
 	}
 }
 
-// AllowConsumerLocked adds the given consumer ID to the Consumable's
-// consumers map. Must be called with Consumable mutex Locked.
-// returns true if changed, false if not
-func (c *Consumable) AllowConsumerLocked(cache *ConsumableCache, id NumericIdentity) bool {
-	consumer := c.getConsumer(id)
-	if consumer == nil {
+// AllowIngressIdentityLocked adds the given security identity to the Consumable's
+// IngressIdentities map. Must be called with Consumable mutex Locked.
+// Returns true if the identity was not present in this Consumable's
+// IngressIdentities map, and thus had to be added, false if it is already added.
+func (c *Consumable) AllowIngressIdentityLocked(cache *ConsumableCache, id NumericIdentity) bool {
+	isIdentityAllowed := c.isIdentityAllowed(id)
+	if isIdentityAllowed == false {
```

Comment:
We still need to differentiate between the cases of an map entry not existing before and one existing with a `false` value. The reason is that all identities are marked as `false` for garbage collection purposes, but we want to return `true` from this function only if a new entry was created, and `false` if we changed the `false` to `true` meaning we are keeping an existing entry. We also do not want to log about a new ingress identity or call `c.addToMaps` for identities that were already allowed.

```
isIdentityAllowed, exists := c.isIdentityAllowed(id)
if !exists {
```

This requires to change `isIdentityAllowed` to return the second boolean return value from the map lookup.

Reply:
I'll just get rid of the function and access the map directly in that case.

Reply:
Fixed.